samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -103,8 +189,38 @@ static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 	}
 }
 
-static void stream_recv_cb(struct bt_bap_stream *stream,
-			   const struct bt_iso_recv_info *info,
+#if defined(CONFIG_LIBLC3)
+static void stream_recv_lc3_codec(struct bt_bap_stream *stream, const struct bt_iso_recv_info *info,
+				  struct net_buf *buf)
+{
+	static uint32_t lc3_recv_cnt;
+	struct broadcast_sink_stream *sink_stream =
+		CONTAINER_OF(stream, struct broadcast_sink_stream, stream);
+
+	lc3_recv_cnt++;
+	if ((lc3_recv_cnt % 1000U) == 0U) {
+		printk("Received %u total ISO packets\n", lc3_recv_cnt);
+	}
+
+	if (lc3_decoder == NULL) {
+		printk("LC3 decoder not setup, cannot decode data.\n");
+		return;
+	}
+
+	if ((info->flags & BT_ISO_FLAGS_VALID) == 0) {
+		printk("Bad packet: 0x%02X\n", info->flags);
```

Comment:
Should we apply the same TX stats for the LC3 recv function as we do for `stream_recv_cb`?

Maybe we should just have a single one, and then only if `CONFIG_LIBLC3=y` then we do 
```

	sink_stream->in_buf = net_buf_ref(buf);
	sink_stream->in_buf_len = buf->len;

	k_work_schedule(&sink_stream->lc3_decode_work, K_NO_WAIT);
```